msu/hooks/ai/tactical/behaviors/ai_throw_bomb.nut

```diff
@@ -0,0 +1,13 @@
+::mods_hookExactClass("ai/tactical/behaviors/ai_throw_bomb", function(o) {
+	local onExecute = o.onExecute;
+	o.onExecute = function( _entity )
+	{
+		local itemsBefore = _entity.getItems().getAllItems();
+		local ret = onExecute(_entity);
+		if (ret && this.m.Skill == null)
+		{
+			this.getItems().payForAction(_entity.getItems().getAllItems().filter(@(idx, item) itemsBefore.find(item) == null));
+		}
+		return ret;
```

Comment:
why can't the other 2 hooks use this method?

Reply:
Because in those cases the total items don't change (aka the entity is not adding/removing items to their slots, only shifting their positions). So I'd instead have to go through each slot and see if the item in it changed instead of just comparing the total items before and total items after. Should I implement going through each slot and checking if it changed? Maybe it is more robust like that? (It will still need special handling for the Disarmed thing but that's ok).

Reply:
Yes, going through a few slots is nothing.

Reply:
Ok I will do it like that.

Reply:
Yeah, this would be much more robust, ideally we should have a single standardized wrapper for this situation.

Reply:
Unresolving this. I've thought about it a bit more and I think having a wrapper for this is unnecessary. If a modder is adding something that swaps items on an AI entity, he can just pass those items to the `payForAction` function in place. The only reason we need to do this comparing thing here is because we want to hook a vanilla function and want to be safe against future changes to that function. So it is a one-off application of this method, which in my opinion has no use for modders.

Reply:
After reading your other comment I agree.